core/src/main/scala/shapeless/MacroState.scala

```diff
@@ -0,0 +1,48 @@
+package shapeless
+
+import scala.reflect.{ClassTag, classTag}
+import scala.reflect.macros.Universe
+
+
+private[shapeless] object MacroState {
```

Comment:
Perhaps this utility could live in macro-compat.

Reply:
macro-compat is on the way out, so it's best for it to be here.